lib/oauth.js

```diff
@@ -558,7 +558,7 @@ internals.encode = function (string) {
 internals.parse = function (payload) {
 
     payload = Buffer.isBuffer(payload) ? payload.toString() : payload;
-    if (payload[0] === '{') {
+    if (/^\s*{/.test(payload)) {
```

Comment:
Yes you're right. For some reason I was assuming nlf was talking about using it destructively a la `payload = payload.trim()`.  No clue why I thought that. I'll switch it to use trim instead.